version: "3"
services:
  server:
    image: sdesbure/domoticz
    healthcheck:
      test: "curl -f http://localhost:8080/ || exit 1"
      interval: 30s
      timeout: 10s
      retries: 3
    networks:
      - proxy
    deploy:
      placement:
        constraints: [node.labels.zwave == true]
      replicas: 1
      labels:
        - com.df.notify=true
        - com.df.distribute=true
        - com.df.serviceDomain=zwave.p.cns.me
        - com.df.servicePath=/
        - com.df.port=8080
        - com.df.xForwardedProto=true
        - com.df.redirectWhenHttpProto=true
    volumes:
      - /dev/bus/usb:/dev/bus/usb
      - config:/config
      - /etc/localtime:/etc/localtime:ro

networks:
  proxy:
    external: true

volumes:
  config:
    driver: glusterfs
